### PR TITLE
JPO: display empty string in form fields instead of `false` when no data is present.

### PR DESCRIPTION
**Pull request by AnnaMag**
The fields in the JPO forms are currently populated with `false` when corresponding options are empty ( which is returned from the JP onboard settings GET endpoint ).  

This patch replaces `false` with empty strings for empty options.

**To test:**
- Checkout this branch on local Calypso.
- Go to `https://YourJetpackSandbox.com/wp-admin/admin.php?page=jetpack&action=onboard&calypso_env=development`, where `YourJetpackSandbox.com` is the domain of your JP sandbox
* Add data to the form in the first screen, press Next Step. 
* On your sandbox delete the corresponding option `wp option delete blogname` and/or  `wp option delete blogdescription`.
* Access the flow again using the above url and verify that the fields with deleted options are empty 
 (and do not display `false`).
* Skip through the flow till the `Add a business address` step.
* Fill-in the required fields, press Next Step.
* Remove the option `wp option delete jpo_business_address` .
* Access url to trigger the flow again, skip to the `Add a business address` step
and verify the fields are empty.

**Note:**
This will only work on initial render (re-accessing the flow via url as done above). Once the state is populated with field data, it renders the content irrespective of whether JP options are empty. Separate issue follows.

### CODE REVIEW

File: client/jetpack-onboarding/steps/business-address.jsx [hunk 1]
```diff
@@ -96,7 +96,7 @@ class JetpackOnboardingBusinessAddressStep extends React.PureComponent {
 									id={ fieldName }
 									onChange={ this.getChangeHandler( fieldName ) }
 									required={ fieldName !== 'state' }
-									value={ this.state[ fieldName ] }
+									value={ this.state[ fieldName ] ? this.state[ fieldName ] : '' }
```
Comment (tyxla) [inline client/jetpack-onboarding/steps/business-address.jsx]: Perhaps we could go with the shorter `this.state[ fieldName ] || ''` here?

File: client/jetpack-onboarding/steps/site-title.jsx [hunk 2]
```diff
@@ -75,8 +75,8 @@ class JetpackOnboardingSiteTitleStep extends React.PureComponent {
 					<form onSubmit={ this.handleSubmit }>
 						<SiteTitle
 							autoFocusBlogname
-							blogname={ this.state.blogname }
-							blogdescription={ this.state.blogdescription }
+							blogname={ this.state.blogname ? this.state.blogname : '' }
```
Comment (tyxla) [inline client/jetpack-onboarding/steps/site-title.jsx]: Perhaps we could go with the shorter `this.state.blogname || ''` here?

File: client/jetpack-onboarding/steps/site-title.jsx [hunk 3]
```diff
@@ -75,8 +75,8 @@ class JetpackOnboardingSiteTitleStep extends React.PureComponent {
...
+							blogdescription={ this.state.blogdescription ? this.state.blogdescription : '' }
```
Comment (tyxla) [inline client/jetpack-onboarding/steps/site-title.jsx]: Perhaps we could go with the shorter `this.state.blogdescription || ''` here?

**Comment by AnnaMag** [comment]
Merging as-is (after the feedback). Subject to the fix of styling changes upon successful form fill-in.

**Pull request merged**